### PR TITLE
Progress indicator docs

### PR DESCRIPTION
**Pull request by Blackbaud-DeniseCisneros**
Phase 2 docs for passive and waterfall indicators. 

Pending demos.

### CODE REVIEW

File: src/app/docs/passive-indicator/passive-indicator-docs.component.html [hunk 1]
```diff
@@ -0,0 +1,153 @@
+<sky-docs-demo-page
+  moduleName="SkyProgressIndicatorModule"
+  moduleSourceCodePath="src/app/public/modules/progress-indicator/"
+  pageTitle="Passive Indicator"
+>
+
+<sky-docs-demo-page-summary>
+  Passive progress indicators represent sequential steps in processes outside of user control.
+</sky-docs-demo-page-summary>
+
```
Comment (blackbaud-johnly) [inline src/app/docs/passive-indicator/passive-indicator-docs.component.html:10]: I assume we don't have the behavioral demos yet, but did you try putting a dummy demo in place just to make sure it works?

File: src/app/docs/passive-indicator/passive-indicator-docs.component.html [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        >
+        </sky-docs-design-guideline-thumbnail>
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline
+        heading="Don't use when"
+        headingStyle="danger"
+      >
+        <p>
+          Don't use passive progress indicators for tasks that require user interaction. Use <a href="https://developer.blackbaud.com/skyux/components/progress-indicator/waterfall-progress-indicator">waterfall progress indicators</a> or <a href="https://developer.blackbaud.com/skyux/components/wizard">wizards</a> instead.
```
Comment (blackbaud-johnly) [inline src/app/docs/passive-indicator/passive-indicator-docs.component.html]: Should the link to the waterfall page be updated to point to the phase 2 docs?

File: src/app/docs/passive-indicator/passive-indicator-docs.component.html [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+        <sky-docs-design-guideline-thumbnail
+          caption="Don't display multiple passive progress indicators in lists."
+          captionType="danger"
+          imageSource="~/assets/img/guidelines/passive-indicator/passive-progress-usage-4.png"
+        >
+        </sky-docs-design-guideline-thumbnail>
+
+        <sky-docs-design-guideline-thumbnail
+          caption="Do roll up steps into a single status in lists. Use a container such as flyouts or popovers to display details from passive progress indicators."
```
Comment (blackbaud-johnly) [inline src/app/docs/passive-indicator/passive-indicator-docs.component.html]: `a container` should be `containers`. ... We also lose the links to the flyout and popover docs, but I don't think we can add links in the captions now. So probably nothing to do about that now.

File: src/app/docs/passive-indicator/passive-indicator-docs.component.html [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+      </sky-docs-anatomy>
+    </sky-docs-demo-page-section>
+
+    <sky-docs-demo-page-section
+      heading="Options"
+    >
+      <sky-docs-design-guideline
+        heading="Step details"
+      >
+        <p>
```
Comment (blackbaud-johnly) [inline src/app/docs/passive-indicator/passive-indicator-docs.component.html]: The paragraphs from here are all too wide for our styles. I'm pretty sure the fix is just to add a class to the `p` tags. I think they all just need `class="sky-docs-text-block"`.

File: src/app/docs/passive-indicator/passive-indicator-docs.component.html [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    </sky-docs-demo-page-section>
+
+    <sky-docs-demo-page-section
+      heading="Related information"
+    >
+      <sky-docs-design-guideline
+        heading="Components"
+      >
+        <ul>
+          <li><a href="https://developer.blackbaud.com/skyux/components/flyout">Flyout</a></li>
```
Comment (blackbaud-johnly) [inline src/app/docs/passive-indicator/passive-indicator-docs.component.html:156]: When I serve this locally, it looks like there is more space between list items than in the other phase 2 docs. But I can't see anything here in the code that would cause the difference.

File: src/app/docs/passive-indicator/passive-indicator-docs.component.html [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+      <sky-docs-design-guideline
+        heading="Guidelines"
+      >
+        <ul>
+          <li><a href="https://developer.blackbaud.com/skyux/design/guidelines/call-out-info">Calling out information</a></li>
+        </ul>
+      </sky-docs-design-guideline>
+    </sky-docs-demo-page-section>
+
+  </sky-docs-design-guidelines>
```
Comment (blackbaud-johnly) [inline src/app/docs/passive-indicator/passive-indicator-docs.component.html:172]: We need to add the JSDoc comments to the components and copy over the existing content on the website (minus `This property accepts...`). I added more detail about this on the toggle switch and key info PRs, but let me know those aren't clear.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 7]
```diff
@@ -0,0 +1,247 @@
+<sky-docs-demo-page
+  moduleName="SkyProgressIndicatorModule"
+  moduleSourceCodePath="src/app/public/modules/progress-indicator/"
+  pageTitle="Waterfall Indicator"
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: The current page title is `Waterfall Page`, but I think this is a good change given that in the design guidelines, we mostly refer to `waterfall progress indicators`. That said, I'd go ahead and add `Progress` in the middle just for consistency with how we refer to this throughout the docs.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+>
+
+<sky-docs-demo-page-summary>
+  Waterfall pages walk users through sequential steps in lengthy or complex tasks.
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I'd change `Waterfall pages` to `Waterfall progress indicators` for consistency with the change in the header. (Unless we decide to switch the header back.)

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+  <sky-docs-design-guidelines>
+
+    <sky-docs-demo-page-section
+      heading="Usage"
+    >
+      <sky-docs-design-guideline
+        heading="Use when"
+        headingStyle="success"
+      >
+        <p>
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html:117]: The paragraphs here are significantly wider than what we have in other docs. But when I check the code for other phase 2 docs like alerts and dropdowns, I don't see any class applied or any other setting that would explain the difference. We might need to ask Steve and/or Todd if either of them know what's going on here.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+        >
+        </sky-docs-design-guideline-thumbnail>
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline
+        heading="Don't use when"
+        headingStyle="danger"
+      >
+        <p>
+          Don’t use waterfall pages when users monitor a sequence of events but are not responsible for completing the steps. Use <a href="https://developer.blackbaud.com/skyux/components/progress-indicator/passive-progress-indicator">passive progress indicators</a> instead.
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I'd change `waterfall pages` here to `waterfall progress indicators`. Other than the page title, first paragraph, and sidebar nav, this is the only other instance of `waterfall page` that I see in these docs.

And speaking of the sidebar nav, when I preview this page and the passive progress indicator phase 2 page, I don't see the nested items in the sidebar nav. Is that just something we are going to lose in phase 2, or is there some setup we need to do to get them back?

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          caption="Don't use waterfall progress indicators for steps outside of a user's control."
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I'd change `a users's control` to `user control` to avoid switching to singular.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+        <sky-docs-design-guideline-thumbnail
+          caption="Don't present waterfall progress indicators before the extent of tasks is known.
+"
+          captionType="danger"
+          imageSource="~/assets/img/guidelines/passive-indicator/waterfall-progress-usage-6.png"
+        >
+        </sky-docs-design-guideline-thumbnail>
+
+        <sky-docs-design-guideline-thumbnail
+          caption="Do present decisions that determine the number and order steps before showing waterfall progress indicators."
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: We are missing `of` between `order` and `steps`.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 13]
```diff
@@ -0,0 +1,247 @@
...
+      </sky-docs-anatomy>
+    </sky-docs-demo-page-section>
+
+    <sky-docs-demo-page-section
+      heading="Options"
+    >
+      <sky-docs-design-guideline
+        heading="Waterfall title"
+      >
+        <p>
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: The paragraphs from here are all too wide for our styles. I'm pretty sure the fix is just to add a class to the `p` tags. I think they all just need `class="sky-docs-text-block"`.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 14]
```diff
@@ -0,0 +1,247 @@
...
+      <sky-docs-design-guideline
+        heading="Step details"
+      >
+        <p>
+          You can provide additional details about completed steps, but limit details to single pieces of information. Don’t go overboard because users can reopen modals to review choices as necessary.
+        </p>
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline
+        heading="Start over action"
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: We are using `start-over` as a compound adjective, so it should really be hyphenated here and in the paragraph that follows.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 15]
```diff
@@ -0,0 +1,247 @@
...
+      >
+        <p>
+          You can include a start over action to let users erase all saved data from the waterfall process. This returns user to the first step or to an earlier point in the workflow if decisions occured before the waterfall progress indicator.
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: In the second sentence, `user` should be `users`.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 16]
```diff
@@ -0,0 +1,247 @@
...
+        </p>
+      </sky-docs-design-guideline>
+    </sky-docs-demo-page-section>
+    
+    <sky-docs-demo-page-section
+      heading="Content"
+    >
+      <sky-docs-design-guideline>
+        <p>
+          Use a waterfall title in place of an ordinary page title to provide context for the overall task. Begin the title with a verb to describe the action that the waterfall steps take when they are all completed, similar to the way modal titles are constructed.
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: It's existing content, but the second sentence is hard to follow. I'd change it to `Follow the same pattern as modal titles and start the waterfall title with a verb that indicates the action that occurs when users complete the waterfall steps.` Or something like that. What do you think?

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 17]
```diff
@@ -0,0 +1,247 @@
...
+        </p>
+        <p>
+          Step labels in waterfall progress indicators start with a number and concisely describe the objective. Use the <b>&lt;verb> &lt;direct object></b> format. In the label for the last step, describe the cumulative change that occurs at the end of the process.
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: Again, this is existing content, but `in waterfall progress indicators` strikes me as unnecessary and redundant. I'd delete it and combine the first and second sentences: `For step labels, start with a number and concisely describe the objective using the **<verb> <direct object>** format.` What do you think?

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 18]
```diff
@@ -0,0 +1,247 @@
...
+        </p>
+        <p>
+          Step action (button) labels closely follow step labels and use the same <b>&lt;verb> &lt;direct object></b> format, but they do not need to be identical. For the first step action label, be generic and conversational (“Get started”), and for completed step actions, use verbs appropriate to post hoc actions (“Change”, “Edit”).
+        </p>
+      </sky-docs-design-guideline>
+
+      <sky-docs-design-guideline
+        heading="Layout"
+      >
+        <p>
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html:308]: The spacing is off again, but I don't see any style applied in other phase 2 docs to enforce the narrower width.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 19]
```diff
@@ -0,0 +1,247 @@
...
+
+    <sky-docs-demo-page-section
+      heading="Related information"
+    >
+      <sky-docs-design-guideline
+        heading="Components"
+      >
+        <ul>
+          <li><a href="https://developer.blackbaud.com/skyux/components/modal">Modal</a></li>
+          <li><a href="https://developer.blackbaud.com/skyux/components/progress-indicator/passive-progress-indicator">Passive progress indicator</a></li>
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: Should the URL for passive progress indicators be updated to point to the new phase 2 docs? Same question for the other links to that topic.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 20]
```diff
@@ -0,0 +1,247 @@
...
+
+      <sky-docs-design-guideline
+        heading="Layout"
+      >
+        <p>
+          Waterfall progress indicators are 400 pixels wide and centered inside the page. The waterfall title is the page title. Do not combine waterfall indicators with other content or actions. If waterfall indicators require extra context such as descriptions, links, or instructions, include it after the waterfall title and before the first step.
+        </p>
+
+        <sky-docs-design-guideline-thumbnail
+          caption="Do use waterfall progress indicators as the only content on a page while the user completes them."
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: To stick with plurals, `the user completes` should be `users complete`.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 21]
```diff
@@ -0,0 +1,247 @@
...
+    </sky-docs-demo-page-section>
+
+    <sky-docs-demo-page-section
+      heading="Related information"
+    >
+      <sky-docs-design-guideline
+        heading="Components"
+      >
+        <ul>
+          <li><a href="https://developer.blackbaud.com/skyux/components/modal">Modal</a></li>
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html:328]: Serving locally, I see more space between the bullets than in other phase 2 docs, but I can't tell what's causing it.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html [hunk 22]
```diff
@@ -0,0 +1,247 @@
...
+      <sky-docs-design-guideline
+        heading="Guidelines"
+      >
+        <ul>
+          <li><a href="https://developer.blackbaud.com/skyux/design/guidelines/form-design">Form design</a></li>
+        </ul>
+      </sky-docs-design-guideline>
+    </sky-docs-demo-page-section>
+
+  </sky-docs-design-guidelines>
```
Comment (blackbaud-johnly) [inline src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html:343]: We need to add the JSDoc comments to the components and copy over the existing content on the website (minus `This property accepts...`). I added more detail about this on the toggle switch and key info PRs, but let me know those aren't clear.

File: src/app/docs/passive-indicator/passive-indicator-docs.component.html [hunk 23]
```diff
@@ -0,0 +1,153 @@
+<sky-docs-demo-page
+  moduleName="SkyProgressIndicatorModule"
+  moduleSourceCodePath="src/app/public/modules/progress-indicator/"
+  pageTitle="Passive Indicator"
```
Comment (blackbaud-johnly) [inline src/app/docs/passive-indicator/passive-indicator-docs.component.html]: The current header has `Progress` in the middle, and I think I'd stick with that. It's how we refer to it everywhere else in this topic.

File: src/app/index.html [hunk 24]
```diff
@@ -1,3 +1,18 @@
+<h2>
+  Docs
+</h2>
+<ul>
+  <li>
+    <a skyAppLink="/docs/passive-indicator">Passive indicator</a>
```
Comment (blackbaud-johnly) [inline src/app/index.html]: Doesn't really matter that much, but I'd add `progress` in the middle here. And same for the next item.

**Review by blackbaud-johnly — CHANGES_REQUESTED** [review]
Sorry for the delay on this one. It got lost in the shuffle somehow. ... It looks like in addition to my feedback, we still need code samples and demos, so we should probably call that out in the next docs check-in. Also, we have some conflicts that need to be resolved. Please let me know if any of my feedback isn't clear or if you need help with anything.

File: src/app/docs/passive-indicator/passive-indicator-docs.component.html (hunk 5 above)
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/passive-indicator/passive-indicator-docs.component.html:156]: I double-checked this and compared it to existing Phase 2 docs, but they looked the same. 🤷

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 21 above)
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html:328]: I double-checked this too against existing phase 2 docs, but I didn't see anything different.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 22 above)
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html:343]: Done! Although, it's worth noting that some fields are still empty because they weren't covered in the current docs, so I'll need a developer to check behind me.

File: src/app/docs/passive-indicator/passive-indicator-docs.component.html (hunk 1 above)
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/passive-indicator/passive-indicator-docs.component.html:10]: I tried adding a demo locally and then serving, and it looks like it's working :)

File: src/app/docs/passive-indicator/passive-indicator-docs.component.html (hunk 5 above)
Comment (blackbaud-johnly) [reply src/app/docs/passive-indicator/passive-indicator-docs.component.html:156]: Very strange. Maybe it'll resolve itself when we merge the docs.

**Review by Blackbaud-SteveBrush — DISMISSED** [review]

File: src/app/public/modules/progress-indicator/progress-indicator.component.ts [hunk 25]
```diff
@@ -70,6 +81,11 @@ export class SkyProgressIndicatorComponent implements OnInit, AfterContentInit,
     return this._isPassive || false;
   }
 
+/**
+ * Specifies an observable of `SkyProgressIndicatorMessageType` that determines the status to
```
Comment (blackbaud-johnly) [inline src/app/public/modules/progress-indicator/progress-indicator.component.ts:99]: I'm not sure what's going on, but when I serve locally, I'm getting the `SkyProgressIndicatorMessageType` text showing up within an `a` tag with `class` and `href` attributes.

image

@Blackbaud-SteveBrush, any idea what is going on here? Looking at this line in the code, the JSDoc comments clearly don't have that `a` tag.

File: src/app/docs/passive-indicator/passive-indicator-docs.component.html (hunk 6 above)
Comment (blackbaud-johnly) [reply src/app/docs/passive-indicator/passive-indicator-docs.component.html:172]: Looks like the developer content is showing up just fine. I'll resolve this and review the individual JSDoc comments.

File: src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts [hunk 26]
```diff
@@ -17,6 +17,10 @@ import {
 })
 export class SkyProgressIndicatorItemComponent {
 
+/**
+ * Specifies a header to display above the progress indicator.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts]: Is this right? I mean, according to the old docs, the `SkyProgressIndicatorTitleComponent` handles the header above the progress indicator. I think we need this description with that component. And I think this property should be something like `Specifies the step label to display for a step in the progress indicator.` I'm using `step label` to coincide with the terminology in the Anatomy graphic in the Design guidelines. Those guidelines also say that you can add step details, but I don't see a property for that. @Blackbaud-SteveBrush, does my description for this `tile` property look right to you? And how/where do you specify step details in the code?

File: src/app/public/modules/progress-indicator/progress-indicator.component.ts (hunk 25 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/progress-indicator/progress-indicator.component.ts:99]: Appears to be a bug. Do you mind filing an issue in docs-tools?

File: src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts (hunk 26 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts]: @blackbaud-johnly You are correct; the replacement description you provided looks good to me. I'm not sure what you mean with your last question, unless you mean the content added within the `sky-progress-indicator-item` component?
```
<sky-progress-indicator-item
  title="This item title"
>
  Details here.
</sky-progress-indicator-item>
```

Comment (blackbaud-johnly) [reply src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts]: My second question is that if we have the `title` property on `sky-progress-indicator-item` to specify the step label, then how/where do users specify the step details? The design guidelines list it as an option, but I don't see a corresponding property in the development guidelines for implementing step details.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs](https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs](https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs)

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 7 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I updated this locally and will push up the change shortly. Resolving the comment ahead of that.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 8 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I updated this locally and will push up the change shortly. Resolving the comment ahead of that.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 9 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html:117]: I dug back in and realized that while I checked the code for other phase 2 docs for classes or something else adjusting the width, I didn't check the live docs. As it turns out, in the live phase 2 docs, the "Use when" and "Don't use when" body copy is wider than in the phase 1 docs. I'm not sure if that's by design, but I'm going to assume so and resolve this conversation.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 10 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I updated this locally and will push up the change shortly. Resolving the comment ahead of that.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 12 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I updated this locally and will push up the change shortly. Resolving the comment ahead of that.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 13 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I updated this locally and will push up the change shortly. Resolving the comment ahead of that.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 14 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I updated this locally and will push up the change shortly. Resolving the comment ahead of that.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 15 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I updated this locally and will push up the change shortly. Resolving the comment ahead of that.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 16 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I updated this locally and will push up the change shortly. Resolving the comment ahead of that.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 18 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html:308]: I think this is just a change from phase 1 to phase 2.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 19 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I updated this locally and will push up the change shortly. Resolving the comment ahead of that.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 17 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I updated this locally and will push up the change shortly. Resolving the comment ahead of that.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 21 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html:328]: No idea what's going on with the spacing when serving locally. I'm going to resolve this comment, and we can create a follow-up issue later if the live docs have inconsistent spacing.

File: src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html (hunk 11 above)
Comment (blackbaud-johnly) [reply src/app/docs/waterfall-indicator/waterfall-indicator-docs.component.html]: I updated this locally and will push up the change shortly. Resolving the comment ahead of that.

File: src/app/public/modules/progress-indicator/progress-indicator.component.ts (hunk 25 above)
Comment (blackbaud-johnly) [reply src/app/public/modules/progress-indicator/progress-indicator.component.ts:99]: I just realized that no one followed up with Steve, and I created a docs-tool issue at https://github.com/blackbaud/skyux-docs-tools/issues/63.

File: src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts (hunk 26 above)
Comment (blackbaud-johnly) [reply src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts]: @Blackbaud-SteveBrush following up with you on the pending question here. I updated the docs for the first item I raised is around how to display step details in the progress indicator. Based on the Anatomy and Options sections of the Design content, consumers can optionally display details under the step label:
image
We have the `title` property on `sky-progress-indicator-item` to specify the step label, but I don't see any property or other input for specifying the optional step details. Am I missing something?

Comment (blackbaud-johnly) [reply src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts]: @Blackbaud-AlexKingman, pinging you on the above question too since I know that you've been helping Denise with these dev properties.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs](https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs](https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs)

File: src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts (hunk 26 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts]: There's not a property for it, but consumers can add additional information by simply including it inside the `sky-progress-indicator-item` tag:
```
      <sky-progress-indicator-item
        title="Finished step"
      >
         Optional details about this step
      </sky-progress-indicator-item>
```

Comment (blackbaud-johnly) [reply src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts]: OK. I see. I'd expected it to be a property alongside the `title` property. I'll serve the docs to check out the description for the item component to see if I was just overlooking the obvious or if we need to tweak its description. Thanks!

File: src/app/public/modules/progress-indicator/progress-indicator.component.ts (hunk 25 above)
Comment (blackbaud-johnly) [reply src/app/public/modules/progress-indicator/progress-indicator.component.ts:99]: Resolving this conversation since it looks like we've got this in a state that avoids the weird `a` tag in the docs.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs](https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs](https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs)

File: src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts (hunk 26 above)
Comment (blackbaud-johnly) [reply src/app/public/modules/progress-indicator/progress-indicator-item/progress-indicator-item.component.ts]: Resolving this because I updated the description around this.

**Review by blackbaud-johnly — DISMISSED** [review]
Still waiting on the waterfall demo, but other than that, everything looks good.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs](https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs)

**Comment by Blackbaud-SomaliaJamall** [comment]
All the demos are in!

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs](https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs](https://developer.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs)

**Review by blackbaud-johnly — DISMISSED** [review]
LGTM. The Stackblitz issue is fixed, so I think the only remaining item to do is to address @Blackbaud-AdamFunderburk's feedback on the centering of the demo. Not sure if that's worth holding up the PR for though since we can always file an issue to handle it later.

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs](https://host.nxt.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs)

**Review by blackbaud-johnly — DISMISSED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs](https://host.nxt.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs)

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs](https://host.nxt.blackbaud.com/progress-indicator?_sp.spa=pr-progress-indicator-docs)

**Review by Blackbaud-AlexKingman — APPROVED** [review]
